### DISCUSSION TITLE
Should we make "info" as the default logging level for PyFluent.

### DISCUSSION THREAD

**Discussion by mkundu1** [ansys/pyfluent#4526] [Ideas]
There are certain informational messages we want to show to PyFluent users (in connection of #2674 and #2865) which are currently categorized as warnings to display them by default. If we make `logging.INFO` as the default level, we will be able to show these message under the correct category. This will mean we need to move the current `logging.INFO` messages which are not relevant to users to `logging.DEBUG` level.

**Comment by raph-luc** [comment]
I considered this as well, I think it could work, but I would say that it would be best to keep it to `WARNING` currently. If you look at how the `logging` package characterizes the [logging levels](https://docs.python.org/3/library/logging.html#logging-levels):

- `DEBUG`: Detailed information, typically only of interest to a developer trying to diagnose a problem.
- `INFO`: Confirmation that things are working as expected.
- `WARNING`: An indication that something unexpected happened, or that a problem might occur in the near future (e.g. ‘disk space low’). The software is still working as expected.
- `ERROR`: Due to a more serious problem, the software has not been able to perform some function.
- `CRITICAL`: A serious error, indicating that the program itself may be unable to continue running.

I think those examples you mentioned there are appropriate at the `WARNING` level: they are indicating that a problem might occur in the future if the user is not aware of these messages.

- For the deprecated warning message, if the user keeps using the deprecated argument in the future, they will eventually run into errors once the deprecated argument is removed (assuming our approach is to mark as deprecated first and then remove, which I think is the intention).

- For the Fluent container message, the user needs to be aware of what paths are being used and the difference between the host paths and the guest container paths, otherwise they will run into issues.

These are not just "confirmations that things are working as expected", so `INFO` isn't quite appropriate in my understanding (if following the package recommended levels).

Another argument against changing to `INFO` by default, is that it gives us less levels and less granularity to work with for the hidden logging levels (only `DEBUG` below it). 

I believe we could also set up our own levels, but I think that would probably be more trouble than it's worth.

**Comment by mkundu1** [comment]
Regarding the deprecated warning message, I was thinking that the second message about how we convert the user input value can be moved to info level, sorry for not being clear.

I think we can discuss further within team.

**Comment by raph-luc** [comment]
BTW after the discussion and to further clarify: switching the default output-to-console level to INFO and keeping DEBUG as the only level that is hidden by default, makes sense to me too, and seems to be the preferred option by the others.

**Comment by hpohekar** [comment]
For https://github.com/ansys/pyfluent/issues/2865 we have chosen it as warning on purpose because warning appears immediately after execution of required code, so user don't need to enable logging or set log level and look into log file.

We can make "info" as the default logging level for PyFluent and if possible it should have same behavior like warning (it should appear immediately after execution of required code).

[a comment by an automated account is not shown]

**Comment by prmukherj** [comment]
Current logging level (WARNING and ERROR) is fine.
